force-app/main/default/classes/TaskTriggerHandler.cls

```diff
@@ -0,0 +1,76 @@
+public without sharing class TaskTriggerHandler {
+  public static void updateRelatedAccounts(List<Task> tasks) {
+    Set<Id> whatIdsRelatedToAccounts = new Set<Id>();
+    for (Task task : tasks) {
+      if (
+        task.WhatId <> null &&
+        task.WhatId.getSobjectType() == Account.getSObjectType()
+      ) {
+        whatIdsRelatedToAccounts.add(task.WhatId);
+      }
+    }
+
+    if (whatIdsRelatedToAccounts.isEmpty()) {
+      return;
+    }
+    List<Account> accounts = countSpecificTasks(
+      getUnitedStatesProspectAccounts(whatIdsRelatedToAccounts)
+    );
+    update accounts;
+  }
+
+  public static List<Account> getUnitedStatesProspectAccounts(Set<Id> whatIds) {
+    Id recordTypeId = Schema.SObjectType.Account.getRecordTypeInfosByDeveloperName()
```

Comment:
if you make this one into a `@testvisible private static final` at the top of your class, you can also use it in your testclass and avoid the same code there.